analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AssignmentExpressionSyntaxExtensionsTests.cs

```diff
@@ -27,38 +27,492 @@ namespace SonarAnalyzer.UnitTest.Extensions
+                        {
+                            Expression = new { Keyword = new { Text = "string" } },
+                            Name = WithIdentifier("Empty"),
+                        },
+                    },
+                    new
+                    {
+                        Left = WithDesignation("d"),
+                        Right = new { Expression = WithIdentifier("M") },
+                    },
+                    new
+                    {
+                        Left = WithIdentifier("_"),
+                        Right = new { Type = new { Keyword = new { Text = "object" } } },
+                    },
+                });
+
         [DataTestMethod]
         // Normal assignment
         [DataRow("int a; a = 1;", "a")]
         // Deconstruction into tuple
-        [DataRow("(var a, var b) = (1, 2);", "var a,var b")]
-        [DataRow("(var a, _) = (1, 2);", "var a,_")]
-        [DataRow("(var _, _) = (1, 2);", "var _,_")]
-        [DataRow("(_, _) = (1, 2);", "_,_")]
-        [DataRow("(var a, (var b, var c), var d) = (1, (2, 3), 4);", "var a,var b,var c,var d")]
-        [DataRow("int b; (var a, (b, var c), _) = (1, (2, 3), 4);", "var a,b,var c,_")]
+        [DataRow("(var a, var b) = (1, 2);", "var a", "var b")]
+        [DataRow("(var a, _) = (1, 2);", "var a", "_")]
+        [DataRow("(var _, _) = (1, 2);", "var _", "_")]
+        [DataRow("(_, _) = (1, 2);", "_", "_")]
+        [DataRow("(var a, (var b, var c), var d) = (1, (2, 3), 4);", "var a", "var b", "var c", "var d")]
+        [DataRow("int b; (var a, (b, var c), _) = (1, (2, 3), 4);", "var a", "b", "var c", "_")]
         // Deconstruction into declaration expression designation
-        [DataRow("var (a, b) = (1, 2);", "a,b")]
+        [DataRow("var (a, b) = (1, 2);", "a", "b")]
         [DataRow("var (a, _) = (1, 2);", "a")]
-        [DataRow("var (_, _) = (1, 2);", "")]
-        public void AssignmentExpressionSyntaxExtensions_AssignmentTargets_DeconstructTargets(string assignment, string expectedTargets)
+        [DataRow("var (_, _) = (1, 2);")]
+        public void AssignmentTargets_DeconstructTargets(string assignment, params string[] expectedTargets)
         {
-            var syntaxTree = CSharpSyntaxTree.ParseText(WrapInMethod(assignment));
-            var assignmentExpression = syntaxTree.GetRoot().DescendantNodesAndSelf().OfType<AssignmentExpressionSyntax>().First();
-            var allTargets = assignmentExpression.AssignmentTargets();
-            var allTargetsAsString = string.Join(",", allTargets.Select(x => x.ToString()));
-            allTargetsAsString.Should().Be(expectedTargets);
+            var allTargets = ParseAssignmentExpression(assignment).AssignmentTargets();
+            var allTargetsAsString = allTargets.Select(x => x.ToString());
+            allTargetsAsString.Should().BeEquivalentTo(expectedTargets);
         }
 
-        private static string WrapInMethod(string code) =>
-$@"
+        #region Helpers
```

Comment:
Not needed, it's obvious from begin `private` and at the end of the file. We're trying to avoid `#region`
```suggestion
```